lib/board.rb

```diff
@@ -29,4 +29,50 @@ def valid_coordinate?(coordinate)
       true
     end
   end
+
+  def valid_placement?(ship, coordinates)
+      if ship.length == coordinates.length && coordinates.all? { |coordinate| valid_coordinate? (coordinate) }
```

Comment:
Notes to myself to remember what it's doing:
This first checks the length where the number of coordinates should equal the length of the ship.
.all? method returns true when the block is true.